truffle/src/main/java/org/jruby/truffle/runtime/object/ObjectGraph.java

```diff
@@ -115,9 +129,23 @@ private void visitObject(DynamicObject object, ObjectGraphVisitor visitor) throw
 
             // Visit all properties
 
-            for (Property property : object.getShape().getProperties()) {
+            for (Property property : object.getShape().getPropertyListInternal(false)) {
```

Comment:
Which HiddenKey do we want to traverse? Before they were fields and only traversed if they were explicitly in visitObjectGraphChildren, right?